frontend/src/pages/EditStudent.vue

```diff
@@ -109,82 +111,68 @@ export default {
+            })
+        // Refresh NativeLanguage
+        this.getNativeLanguages()
+        setTimeout(() => {
+          this.$router.push(
+            {path: `/students/${this.studentData.StudentID}`})}, 3000)
+      } 
+      else {
+         this.$buefy.notification.open({ 
+              message: 'Something went wrong. Please try again.',
+              duration: 3000, 
+              type: 'is-warning',
+              position: 'is-top'
+            })
+      }
     }
   },
-
- async mounted() {
+  async mounted(){
+   
     // Call API for Native Languages
-      await fetch("/api/nativeLanguages")
-        .then(response => response.json())
-        .then(result => this.API_nativeLanguage = result)
-      
+      this.getNativeLanguages()
     // Call API for Student data
       const id = this.$route.params.id
-      await fetch(`/api/students/${id}`)
-      .then(response => response.json())
-      .then(result => {
-        //transform Nativelanguage ID from int to string 
-        const value = {
+      const result = this.getStudentByStudentId(id)
+      this.studentData = {
           FullName: result.FullName,
           PhoneNumber: result.PhoneNumber,
           Source: result.Source,
           EnglishProficiency: result.EnglishProficiency,
           Notes: result.Notes,
-          NativeLanguageString: result.nativeLanguage.NativeLanguage
+          nativeLanguage: result.nativeLanguage.NativeLanguage,
+          StudentID: result.StudentID
         }
-  
-
-        this.studentData = value;
-        });
+      this.selected.NativeLanguage = result.nativeLanguage.NativeLanguage
   },
 
   methods: {
+    ...mapActions(['getNativeLanguages', 'patchStudent']),
     saveStudent(){
-      // HTTP PATCH to update student, changing specific field on backend.
-        const updateData = {...this.studentData}
-        const studentSave = {
-        method: "PATCH",
-        headers: {
-          'Accept': 'application/json',
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify(
-          updateData 
-        )
-      }
-      const id = this.$route.params.id
-      fetch(`/api/students/${id}`, studentSave)
-        .then(response => {
-          if (response.status < 400) {
-            this.$buefy.notification.open({
-              message: 'Student saved. <u>View profile</u>!',
-              duration: 3000,
-              type: 'is-success',
-              position: 'is-top',
-            })
-            setTimeout(() => {this.$router.push({path: `/students/${id}`})}, 5000)} 
-          else {
-            this.$buefy.notification.open({ 
-              message: 'Something went wrong. Please try again.',
-              duration: 3000, 
-              type: 'is-warning',
-              position: 'is-top'
-            })
-          }
-       })
+        const updateData = {...this.studentData, NativeLanguageString: this.selected.NativeLanguage}
+        this.patchStudent(updateData)
```

Comment:
```suggestion
        this.patchStudent(updateData).then(() => {
          if(this.myUpdateStudentSuccess){
            this.$buefy.notification.open({
                  message: 'Student saved. <u>View profile</u>!',
                  duration: 3000,
                  type: 'is-success',
                  position: 'is-top',
                })
            // Refresh NativeLanguage
            this.getNativeLanguages()
            setTimeout(() => {
              this.$router.push(
                {path: `/students/${this.studentData.StudentID}`})}, 3000)
          }
          else {
            this.$buefy.notification.open({
                  message: 'Something went wrong. Please try again.',
                  duration: 3000,
                  type: 'is-warning',
                  position: 'is-top'
                })
          }
        })
```

Move notification display into `saveStudent()`, so that the notification will pop up whether or not the success status changes